crates/web/src/components/publisher/mod.rs

```diff
@@ -1,21 +1,41 @@
-use leptos::{component, create_action, create_rw_signal, view, IntoView, SignalGet};
+use leptos::{
+    component, create_action, create_rw_signal, view, IntoView, RwSignal, Show, SignalGet,
+    SignalSet,
+};
 use townhall_client::{post::post_create::post_create::PostCreateInput, Client};
 
 use crate::components::text_field::TextField;
 
 #[component]
 pub fn Publisher() -> impl IntoView {
+    let text_title = create_rw_signal(String::default());
     let text_content = create_rw_signal(String::default());
-    let send_post_action = create_action(|content: &String| {
-        let content = content.to_owned();
+
+    let creation_error: RwSignal<Option<String>> = create_rw_signal(None);
+
+    let send_post_action = create_action(move |data: &(String, String)| {
+        let (title, content) = data;
+
+        let title = title.trim().to_owned();
+
+        if title.is_empty() {
+            creation_error.set(Some("Title is required".to_owned()));
+            ()
```

Comment:
Perhaps you wanted to do this instead?

```suggestion
            return;
```